File: kotlin_android_dev/app/src/main/kotlin/com/kotlin_android_dev/model/example/ListExample.kt
package com.kotlin_android_dev.model.example

import io.github.antonpichka.lamm.base_model.BaseListModel

open class ListExample<T : Example>(listModel: MutableList<T>) : BaseListModel<T>(listModel) {
    @Suppress("UNCHECKED_CAST")
    override fun clone(): ListExample<T> {
        val newListModel = mutableListOf<T>()
        for(itemModel: T in listModel) {
            newListModel.add(itemModel.clone() as T)
        }
        return ListExample(newListModel)
    }

    override fun toString(): String {
        var strListModel = "\n"
        for(itemModel: T in listModel) {
            strListModel += "$itemModel,\n"
        }
        return "ListExample(listModel: [$strListModel])"
    }
}

File: dart_web_dev/app/lib/models/example/list_example.dart
import 'package:dart_web_dev/models/example/example.dart';
import 'package:meta/meta.dart';
import 'package:smvp_dart/smvp.dart';

@immutable
base class ListExample<T extends Example> extends BaseListModel<T> {
  const ListExample({required super.listModel});

  @override
  ListExample<T> clone() {
    List<T> newListModel = List.empty(growable: true);
    for (final T itemModel in listModel) {
      newListModel.add(itemModel.clone() as T);
    }
    return ListExample<T>(listModel: newListModel);
  }

  @override
  List<Map<String, dynamic>> toListMap() {
    final List<Map<String, dynamic>> listMap = List.empty(growable: true);
    for (final T itemModel in listModel) {
      listMap.add(itemModel.toMap());
    }
    return listMap;
  }

  @override
  String toString() {
    String str = "\n";
    for (final T itemModel in listModel) {
      str += "$itemModel,\n";
    }
    return "ListExample(listModel: [$str])";
  }
}
